cli/config/config.go

```diff
@@ -0,0 +1,212 @@
+package config
+
+import (
+	"errors"
+	"fmt"
+	"github.com/up9inc/mizu/cli/mizu"
+	"github.com/up9inc/mizu/cli/uiUtils"
+	"gopkg.in/yaml.v3"
+	"io/ioutil"
+	"os"
+	"path"
+	"reflect"
+	"strconv"
+	"strings"
+)
+
+const separator = "="
+
+var configObj = map[string]interface{}{}
+
+type CommandLineFlag struct {
+	CommandLineName   string
+	YamlHierarchyName string
+	DefaultValue      interface{}
+	Type              reflect.Kind
+}
+
+var CommandLineValues []string
+
+const (
+	ConfigurationKeyAnalyzingDestination = "tap.dest"
+	ConfigurationKeyUploadInterval       = "tap.uploadInterval"
+	ConfigurationKeyMizuImage            = "tap.mizuImage"
+)
+
+var allowedSetFlags = []CommandLineFlag{
+	{
+		CommandLineName:   "dest",
+		YamlHierarchyName: ConfigurationKeyAnalyzingDestination,
+		DefaultValue:      "up9.app",
+		Type:              reflect.String,
+		// TODO: maybe add short description that we can show
+	},
+	{
+		CommandLineName:   "uploadInterval",
+		YamlHierarchyName: ConfigurationKeyUploadInterval,
+		DefaultValue:      10,
+		Type:              reflect.Int,
+	},
+	{
+		CommandLineName:   "mizuImage",
+		YamlHierarchyName: ConfigurationKeyMizuImage,
```

Comment:
can we remove the prefix "Configuration" from all YamlHierarchyName?

Reply:
I think this is a good convention to understand those consts are configuration keys.. maybe I can do it with Enum or something..